src/api/procedures/setTransferRestrictions.ts

```diff
@@ -66,32 +76,59 @@ export async function prepareSetTransferRestrictions(
     context: {
       polymeshApi: {
         tx: { statistics },
+        query: { asset },
         consts,
       },
     },
-    storage: {
-      restrictionsToAdd,
-      restrictionsToRemove,
-      exemptionsToAdd,
-      exemptionsToRemove,
-      occupiedSlots,
-    },
+    storage: { currentRestrictions, occupiedSlots, needStat, currentStats },
+    context,
   } = this;
   const {
     restrictions: { length: newRestrictionAmount },
+    restrictions,
+    type,
+    ticker,
   } = args;
 
-  const restrictionsToAddAmount = restrictionsToAdd.length;
-  const restrictionsToRemoveAmount = restrictionsToRemove.length;
-  const exemptionsToAddAmount = exemptionsToAdd.length;
-  const exemptionsToRemoveAmount = exemptionsToRemove.length;
-
-  if (
-    !restrictionsToAddAmount &&
-    !restrictionsToRemoveAmount &&
-    !exemptionsToAddAmount &&
-    !exemptionsToRemoveAmount
-  ) {
+  const rawTicker = stringToTicker(ticker, context);
+
+  const exemptions: (string | Identity)[] = [];
+
+  let someDifference = restrictions.length !== currentRestrictions.length;
+  const conditions = restrictions
+    .map(r => {
+      if (!someDifference) {
+        someDifference = !currentRestrictions.find(transferRestriction => {
+          if (transferRestriction.isMaxInvestorCount && type === TransferRestrictionType.Count) {
+            const currentCount = u64ToBigNumber(transferRestriction.asMaxInvestorCount);
+            return currentCount.eq((r as CountTransferRestrictionInput).count);
+          } else if (transferRestriction.isMaxInvestorOwnership && type === 'Percentage') {
+            const currentOwnership = permillToBigNumber(transferRestriction.asMaxInvestorOwnership);
+            return currentOwnership.eq((r as PercentageTransferRestrictionInput).percentage);
+          }
+          return false;
+        });
+      }
+      let condition: TransferRestriction;
+      if (type === TransferRestrictionType.Count) {
+        condition = { type, value: (r as CountTransferRestrictionInput).count };
+      } else {
+        condition = { type, value: (r as PercentageTransferRestrictionInput).percentage };
+      }
+
+      const rawCondition = transferRestrictionToPolymeshTransferCondition(condition, context);
+
+      if (r.exemptedIdentities) {
+        r.exemptedIdentities.forEach(e => {
+          exemptions.push(e);
```

Comment:
This isn't 100% accurate, since exemptions are based on type, and don't reference a particular condition.

That said, I think with claim scopes this might make more sense so I am not sure if we should change it.

<img width="930" alt="Screen Shot 2022-04-28 at 10 29 34 PM" src="https://user-images.githubusercontent.com/86971143/165876952-6a2c1fb5-c383-4580-a269-b22063200599.png">

Reply:
https://polymath.atlassian.net/browse/DA-317